service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/NoticeOfProceedingsController.java

```diff
@@ -104,17 +104,45 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmitEvent(
             .getNoticeOfProceedingTemplateData(caseData);
 
         List<DocmosisTemplates> templateTypes = getProceedingTemplateTypes(caseData);
-        
+
         List<Document> uploadedDocuments = generateAndUploadDocuments(userId, authorization, templateData,
             templateTypes);
 
-        caseDetails.getData().put("noticeOfProceedingsBundle", createNoticeOfProceedingsCaseData(uploadedDocuments));
+        List<Element> noticeOfProceedingCaseData = createNoticeOfProceedingsCaseData(uploadedDocuments);
+
+        if (callbackRequest.getCaseDetailsBefore() != null
+            && callbackRequest.getCaseDetailsBefore().getData().get("noticeOfProceedingsBundle") != null) {
+            CaseData caseDataBefore = mapper.convertValue(callbackRequest.getCaseDetailsBefore().getData(),
+                CaseData.class);
+
+            noticeOfProceedingCaseData.addAll(getRemovedDocumentBundles(caseDataBefore, templateTypes));
+        }
+
+        caseDetails.getData().put("noticeOfProceedingsBundle", noticeOfProceedingCaseData);
```

Comment:
Couldn't that be extracted to the service? The integration tests would be easier for you and you could focus more on unit testing.

Reply:
Extracted to NoticeOfProceedingService